atom/browser/api/atom_api_web_contents.cc

```diff
@@ -789,8 +789,11 @@ void WebContents::AddNewContents(content::WebContents* source,
     if (disposition != WindowOpenDisposition::NEW_WINDOW &&
         disposition != WindowOpenDisposition::NEW_POPUP) {
       if (!browser) {
-        // TODO(bridiver) - this should be the source owner_window
-        browser = owner_window()->browser();
+        extensions::TabHelper* source_tab_helper =
+              extensions::TabHelper::FromWebContents(source);
+        CHECK(source_tab_helper);
```

Comment:
This isn't the right thing to do here. In some cases the main window is the source or an extension background page can be the source. In those cases we should fall back to the owner_window